Kitodo/src/main/java/org/kitodo/production/services/file/MetadataImageComparator.java

```diff
@@ -34,13 +34,6 @@ public int compare(Object firstObject, Object secondObject) {
         if (firstObject instanceof String && secondObject instanceof String) {
             firstName = (String) firstObject;
             secondName = (String) secondObject;
-
-            int firstNameLastPeriodIndex = firstName.lastIndexOf('.');
-            int secondNameLastPeriod = secondName.lastIndexOf('.');
-            if (firstNameLastPeriodIndex > -1 && secondNameLastPeriod > -1) {
-                firstName = firstName.substring(0, firstNameLastPeriodIndex);
-                secondName = secondName.substring(0, secondName.lastIndexOf('.'));
-            }
```

Comment:
I am not sure if this part should really be removed. Apparently file extensions are not to be considered when comparing images (maybe the `compare` method is meant to be invariant of invidual media variants/represenstations of one image/scan?)

@matthias-ronge even though this part of the code is quite old, do you remember why you stripped the file extension from the filenames before passing the parameters to the `compareImages` method?

Reply:
This originates from 2.x, though [there](/kitodo/kitodo-production/blob/2.x/Goobi/src/de/sub/goobi/metadaten/MetadatenImagesHelper.java#L445) the code did read quite different. In case that the config parameter `image.sorting` is set to `number`, the images are compared by their integer value (`Integer.valueOf()`) of their file name, so the ending must be cut off to allow conversion to `int`.

Reply:
It seems to me that the stripping of the file extension is already happening earlier, when searchForMedia is called so that the removal of those lines might not have an effect in the code section discussed here.

https://github.com/kitodo/kitodo-production/blob/3ca7384e44f8681e9b49de91738ee05e2333584b/Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java#L1099-1104

Reply:
It seems to me that for the problem described in the linked issue the change in FileService:

```java
public String getFileName(URI uri) {
        String fileNameWithExtension = fileManagementModule.getFileNameWithExtension(uri);
        if (fileNameWithExtension.contains(".")) {
            return fileNameWithExtension.substring(0, fileNameWithExtension.lastIndexOf('.'));
        }
        return fileNameWithExtension;
    }
```
is not even relevant. The change which fixes the problem in the linked issue is the deletion of lines in the `MetadataImageComparator` because the file extension are already stripped of here as it seems to me while the files are collected.

https://github.com/kitodo/kitodo-production/blob/7e2caad5fdbf06929bf53418e86dd80b93c3ed35/Kitodo/src/main/java/org/kitodo/production/model/Subfolder.java#L346-L355

Reply:
In order to prevent the corruption of file name with dots, the convention should probably be the following: if a String is passed to the comperator, this String is always a filename without  an extension in order to not having to implement the removal of extensions in the Comparator which has no knowledge what kind of String is given to it.